packages/check/match.js

```diff
@@ -266,17 +287,29 @@ var checkSubtree = function (value, pattern) {
   }
 
   if (typeof pattern !== "object")
-    throw Error("Bad pattern: unknown pattern type");
+    return {
+      message: "Bad pattern: unknown pattern type",
+      path: ""
+    };
 
   // An object, with required and optional keys. Note that this does NOT do
   // structural matches against objects of special types that happen to match
   // the pattern: this really needs to be a plain old {Object}!
   if (typeof value !== 'object')
-    throw new Match.Error("Expected object, got " + typeof value);
+    return {
+      message: "Expected object, got " + typeof value,
+      path: ""
+    };
   if (value === null)
-    throw new Match.Error("Expected object, got null");
+    return {
```

Comment:
Since these if statements have become multiple lines, let's wrap the body in brackets.